test-integration/src/test/java/nl/nn/adapterframework/filesystem/AmazonS3FileSystemTestHelper.java

```diff
@@ -183,7 +188,23 @@ public boolean _folderExists(String folderName) throws Exception {
 	@Override
 	public void _deleteFolder(String folderName) throws Exception {
 		String foldername = folderName.endsWith("/") ? folderName : folderName + "/";
-		_deleteFile(null, foldername);
+		ObjectListing objectListing = s3Client.listObjects(bucketName, foldername);
+		while (true) {
+			Iterator<S3ObjectSummary> objIter = objectListing.getObjectSummaries().iterator();
+			while (objIter.hasNext()) {
+				s3Client.deleteObject(bucketName, objIter.next().getKey());
+			}
+
+			// If the bucket contains many objects, the listObjects() call
+			// might not return all of the objects in the first listing. Check to
+			// see whether the listing was truncated. If so, retrieve the next page of objects 
+			// and delete them.
+			if (objectListing.isTruncated()) {
+				objectListing = s3Client.listNextBatchOfObjects(objectListing);
+			} else {
+				break;
+			}
+		}
```

Comment:
This whole loop seems duplicated with the teardown-method. Perhaps a good idea to extract the duplicate code into a method.